sycl/doc/extensions/proposed/sycl_ext_oneapi_annotated_ptr.asciidoc

```diff
@@ -0,0 +1,655 @@
+loads from and stores to the pointers.
+
+```c++
+namespace sycl::ext::oneapi {
+template <typename T, typename PropertyListT = properties<>>
+class annotated_ref {
+  public:
+    annotated_ref(T *);
+    operator T() noexcept;
+    operator const T() const noexcept;
+    void operator=(const T &);
+  private:
+    T *Ptr;
+  };
+} // namespace sycl::ext::oneapi
+```
+
+
+Member Functions are described in the table below
+[frame="topbot",options="header"]
+|===
+|Functions |Description
+
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+annotated_ref(T * InputPtr);
+----
+|
+Constructs an `annotated_ref` object. Does not allocate new storage. The
+underlying pointer is initialized to `InputPtr`.
+
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+operator T() noexcept;
+operator const T() const noexcept;
+----
+|
+Reads the object of type `T` that is referenced by this wrapper, applying the
+annotations when the object is loaded from memory.
+
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+void operator=(const T &);
+----
+|
+Writes an object of type `T` to the location referenced by this wrapper,
+applying the annotations when the object is stored to memory.
+
+|===
+
+=== Properties
+
+Below is a list of compile-time constant properties supported with
+`annotated_ptr`. 
+
+```c++
+namespace sycl::ext::oneapi::experimental {
+struct alignment_key {
+  template<int K>
+  using value_t = property_value<alignment_key, std::integral_constant<int, K>>;
+};
+
+struct restrict_key {
+  using value_t = property_value<restrict_key>;
+};
+
+struct runtime_aligned_key {
+  using value_t = property_value<runtime_aligned_key>;
+};
+
+template<int K>
+inline constexpr alignment_key::value_t<K> alignment;
+inline constexpr restrict_key::value_t restrict;
+inline constexpr runtime_aligned_key::value_t runtime_aligned;
+
+template<>
+struct is_property_key<alignment_key> : std::true_type {};
+template<>
+struct is_property_key<restrict_key> : std::true_type {};
+template<>
+struct is_property_key<runtime_aligned_key> : std::true_type {};
+
+template<typename T, typename PropertyListT>
+struct is_property_key_of<
+  alignment_key, annotated_ptr<T, PropertyListT>> : std::true_type {};
+template<typename T, typename PropertyListT>
+struct is_property_key_of<
+  restrict_key, annotated_ptr<T, PropertyListT>> : std::true_type {};
+template<typename T, typename PropertyListT>
+struct is_property_key_of<
+  runtime_aligned_key, annotated_ptr<T, PropertyListT>> : std::true_type {};
+} // namespace experimental::oneapi::ext::sycl
+```
+--
+[options="header"]
+|====
+| Property | Description
+|`alignment`
+|
+Sets the alignment of the pointer address in bytes.
+
+|`restrict`
+|
+Informs the compiler that writes to the address pointed to by this pointer
+are only done by this pointer or pointers derived from it.
+
+|`runtime_aligned`
+|
+Informs the compiler that the pointer has the alignment as determined by the
```

Comment:
This wording is quite vague.  How about this:

Informs the compiler that the pointer has at least the default alignment for a pointer allocated through the SYCL runtime.  This is always safe to apply to any pointer returned by a SYCL memory allocation function.